### DISCUSSION TITLE
Can loadBinary be moved to be a loadPyodide option?

### DISCUSSION THREAD

**Discussion by szabolcsdombi** [pyodide/pyodide#3944] [Q&A]
Hi, I am interested to hook the fetch calls made by pyodide.

I found:

```js
/** @private */
export let loadBinaryFile: (
  path: string,
  file_sub_resource_hash?: string | undefined,
) => Promise<Uint8Array>;
if (IN_NODE) {
  loadBinaryFile = node_loadBinaryFile;
} else {
  loadBinaryFile = browser_loadBinaryFile;
}
```

https://github.com/pyodide/pyodide/blob/main/src/js/compat.ts#L160-L169

I am interested if this method could be overwritten by the options in loadPyodide().
Right now there is no such member in the options.

I would like to add it myself, but I am not sure if such a thing would be welcome.
I am also not sure if this is the right way to do it.

My current workaround is to replace the window.fetch.

**Comment by hoodmane** [comment]
This seems like a fine feature, I don't see why not.

**Comment by ryanking13** [comment]
I would like to implement it with a hook system (#3316), so people can modify other methods not just `loadBinary`. But I think it won't be implemented in the short term, so I am OK with adding it to the loadPyodide option as a workaround.
